browser/brave_shell_integration.cc

```diff
@@ -11,12 +11,39 @@
 
 #include <utility>
 
+#include "brave/browser/brave_shell_integration_win.h"
 #include "brave/browser/default_protocol_handler_utils_win.h"
 #include "chrome/installer/util/shell_util.h"
 #endif
 
+#if BUILDFLAG(IS_MAC)
+#include "brave/browser/brave_shell_integration_mac.h"
+#endif
+
 namespace shell_integration {
 
+void PinShortcut(Profile* profile,
+                 base::OnceCallback<void(bool)> result_callback) {
+#if BUILDFLAG(IS_WIN)
+  win::PinToTaskbar(profile, std::move(result_callback));
+#elif BUILDFLAG(IS_MAC)
+  // Mac doesn't support profile specific icon in dock.
+  mac::AddIconToDock(std::move(result_callback));
+#elif BUILDFLAG(IS_LINUX)
+  // Not supported on linux yet.
```

Comment:
`NOTIMPLEMENTED()` or `NOTREACHED()`.

Reply:
Added `NOTREACHED()`

Reply:
can be `NOTREACHED() << "Not supported on Linux yet.";` 😉

Reply:
Done :) Thanks!